sdks/java/core/src/main/java/org/apache/beam/sdk/testing/TestPipeline.java

```diff
@@ -107,7 +107,7 @@
 
   private final PipelineOptions options;
 
-  private static class PipelineRunEnforcement {
+  protected static class PipelineRunEnforcement {
```

Comment:
Could these be package-private instead?
```suggestion
  static class PipelineRunEnforcement {
```

Reply:
They can be, it could be argued that `protected` allows for some extensibility, i.e., if they were protected to begin with, I could have added a subclass implementing the 2 JUnit5 interfaces somewhere in our own test code instead of making a whole Beam PR.

I don't know if there's an **actual** use case mind you, I'm unfamiliar with TestNG and other test frameworks and whether there's any need, so perfectly happy to make them package-private to hide from clients.

Reply:
Noting that you can always extend, in probably a better way, via object composition. You can subclass `Pipeline` and forward method calls to a local `TestPipeline` variable.